scene/gui/tree.cpp

```diff
@@ -2297,11 +2319,8 @@ int Tree::draw_item(const Point2i &p_pos, const Point2 &p_draw_ofs, const Size2
 
 				} break;
 				case TreeItem::CELL_MODE_CUSTOM: {
-					if (p_item->cells[i].custom_draw_obj.is_valid()) {
-						Object *cdo = ObjectDB::get_instance(p_item->cells[i].custom_draw_obj);
-						if (cdo) {
-							cdo->call(p_item->cells[i].custom_draw_callback, p_item, Rect2(item_rect));
-						}
+					if (p_item->cells[i].custom_draw_callback.is_valid()) {
+						p_item->cells[i].custom_draw_callback.call(p_item, Rect2(item_rect));
```

Comment:
If the provided callback can't be called (due to argument mismatch etc.), this will fail silently. We could print an error when that happens.

Though it's a pre-existing problem, so it can be fixed later.

Reply:
I'll take a look at fixing it here 🙂

Reply:
Attempted an improvement of this part